### PR TITLE
modification to resnet_model.py and resnet_main.py

### PR DESCRIPTION
**Pull request by ultrons**
resnet_model.py:
- Replaced _batch_norm function with tf.layers.batch_normalization
It was also one of TODOs as code comment.

- resnet_model.py, resnet_main.py
Added hyperparameter options for dropout on identity and conv branch going to addition block with residual cell. As also noted in Wide Resnet Paper (https://arxiv.org/pdf/1605.07146.pdf) that dropout was found to be counter-productive on the identity branch. Authors recommend to implement it between conv layers instead. in my experiment however dropout on conv branch just before the addition actually improves test accuracy (92.98 in ~70.0k steps, keep_prob=0.9). I thought it could useful for other explorers.

### CODE REVIEW

**Comment by ultrons** [comment]
shifting and scaling operations in during batch normalization for plain fully connected layer use normalization across the batch to compute shift and offset values. In case of conv-nets, I think, the reason why normalizing across all the dimensions but the channels is most widely used is because variation across channel is a characteristic calculated from different filters (distinct in terms of filter parameters). I wanted to experiment what happens if we treat multiple instances of same filter as we convolve also as distinct. Will it really help or harm the learning?

Results were almost the same, I am working on contrasting the two approaches. In favor of implementation simplicity, I moved to tf.layers.batch_normalization. The distinct advantage of which in my experience is that your code will be less error prone. It is fun to debug,  thanks to tensorboard, if you forgot to add op calculating running mean and running variance as training op, but you lose time.

**Comment by Kongsea** [comment]
This [PR](https://github.com/tensorflow/models/pull/1342) has fixed this TODO previously.

**Comment by tfboyd** [comment]
Closing very old PRs.  I do not like seeing work lost.  I and others are working to accept any changes to /official; but changes to other directories require the original owner to take action.  I am sorry this change died/stalled.

I am copy and pasting this message in each PR I a closing (yes kind of like an automated response); and I want to say I feel this way about each PR and wish they had ended with a merge or some other more positive finality.  

Thank you for your contribution and time.
